pkg/templates/nstemplatetiers/nstemplatetier_generator.go

```diff
@@ -2,24 +2,28 @@ package nstemplatetiers
 
 import (
 	"context"
+	"embed"
+	"fmt"
+	"strings"
 
-	"github.com/codeready-toolchain/host-operator/pkg/templates/assets"
+	"github.com/codeready-toolchain/host-operator/pkg/templates"
 	commonclient "github.com/codeready-toolchain/toolchain-common/pkg/client"
 	"github.com/codeready-toolchain/toolchain-common/pkg/template/nstemplatetiers"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
-
 	"github.com/pkg/errors"
 	"gopkg.in/yaml.v2"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const NSTemplateTierRootDir = "templates/nstemplatetiers"
+
 // CreateOrUpdateResources generates the NSTemplateTier resources from the cluster resource template and namespace templates,
 // then uses the manager's client to create or update the resources on the cluster.
-func CreateOrUpdateResources(ctx context.Context, s *runtime.Scheme, client runtimeclient.Client, namespace string, assets assets.Assets) error {
+func CreateOrUpdateResources(ctx context.Context, s *runtime.Scheme, client runtimeclient.Client, namespace string, nstemplatetierFS embed.FS, root string) error {
```

Comment:
The `nstemplatetierFS` and `root` always use the same values apart from 2 tests. Could we do away with those parameters (ideally, without sacrificing the tests either)?

Reply:
i am not sure i understand it correctly, do you mean to remove these parameters?

Reply:
Yes, I think ideally the `nstemplatetiersFS` and `root` params should be removed because in the runtime code they always have the same value.

Reply:
Okay so  i removed the parameters from the main function and and created an another sub func added the parameters there so that i can use it for test and still it addresses the above concern.
PTAL @metlos .